From 8fa28ade90f5b453e92ff828a45563132268911a Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 3 Jul 2017 16:26:12 +0200
Subject: [PATCH 1/1] efi_loader: write reboot reason

In efi_reset_system_boottime output the reason for reboot to the
console.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_runtime.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/efi_loader/efi_runtime.c b/lib/efi_loader/efi_runtime.c
index dd52755d1d..72044cb9b6 100644
--- a/lib/efi_loader/efi_runtime.c
+++ b/lib/efi_loader/efi_runtime.c
@@ -74,8 +74,15 @@ static void EFIAPI efi_reset_system_boottime(
 			efi_status_t reset_status,
 			unsigned long data_size, void *reset_data)
 {
+	uint16_t new_line[] = {0x000d, 0x000a, 0x0000};
+
 	EFI_ENTRY("%d %lx %lx %p", reset_type, reset_status, data_size,
 		  reset_data);
+	if (data_size) {
+		efi_cout_output_string(&efi_con_out, new_line);
+		efi_cout_output_string(&efi_con_out, reset_data);
+		efi_cout_output_string(&efi_con_out, new_line);
+	}
 
 	switch (reset_type) {
 	case EFI_RESET_COLD:
-- 
2.11.0

